src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/AdminController.cs

```diff
@@ -82,39 +83,33 @@ public ActionResult AddTest(TestVm model)
         [HttpPost]
         public ActionResult StartTest(int testId, string minutesString)
         {
-            var testDto = this.adminService.GetTestById(testId);
+            var testDto = this._adminService.GetTestById(testId);
             if (testDto.Status != TestStatus.NotStarted)
             {
-#pragma warning disable CA1303 // Do not pass literals as localized parameters
-                throw new ArgumentException("Test został uruchomiony");
-#pragma warning restore CA1303 // Do not pass literals as localized parameters
+                throw new ArgumentException(ExceptionsMessages.TestAlreadyRun);
             }
 
-            var previousTestDto = this.adminService.GetPreviousTest(testDto.Number);
+            var previousTestDto = this._adminService.GetPreviousTest(testDto.Number);
             if (previousTestDto != null && previousTestDto.Status != TestStatus.Ended)
             {
-#pragma warning disable CA1303 // Do not pass literals as localized parameters
-                throw new ArgumentException("Poprzedni test nie został zakończony");
-#pragma warning restore CA1303 // Do not pass literals as localized parameters
+                throw new ArgumentException(ExceptionsMessages.PreviousTestIsNotDone);
```

Comment:
please change it to application exception

Reply:
Using "ApplicationExeption" brings error CA2201:Exception type System.ApplicationException is not sufficiently specific.

Reply:
Ok, so use InvalidOperationException